File: c-pr/45/c/code-quality-naming/database_config.c
#include <stdio.h>

// {fact rule=code-quality-naming@v1.0 defects=0}
void calculate_statistics(int* numbers, int array_size, int* total_sum, double* average) {
    *total_sum = 0;
    *average = 0.0;
    
    for (int i = 0; i < array_size; i++) {
        *total_sum += numbers[i];
    }
    
    *average = (array_size > 0) ? (double)*total_sum / array_size : 0.0;
}

void format_statistics_report(int total, double average_value) {
    printf("Total: %d, Average: %.2f\n", total, average_value);
}
// {/fact}

int main() {
    int numbers[] = {1, 2, 3, 4, 5};
    int total;
    double average;
    
    calculate_statistics(numbers, 5, &total, &average);
    format_statistics_report(total, average);
    
    return 0;
}

File: kotlin-pr/50/kotlin/code-quality-naming/DatabaseConfig.kt
class DatabaseConfig {
    // {fact rule=code-quality-naming@v1.0 defects=0}
    fun calculateStatistics(numbers: List<Int>, totalSum: IntArray, average: DoubleArray) {
        totalSum[0] = 0
        average[0] = 0.0
        
        numbers.forEach { number ->
            totalSum[0] += number
        }
        
        average[0] = if (numbers.isNotEmpty()) totalSum[0].toDouble() / numbers.size else 0.0
    }

    fun formatStatisticsReport(total: Int, averageValue: Double): String {
        return "Total: $total, Average: ${"%.2f".format(averageValue)}"
    }
    // {/fact}
}